### PR TITLE
🐛 sku api should use resourceType, not kind

### PR DESCRIPTION
**Pull request by alexeldeib**
Signed-off-by: Alexander Eldeib <alexeldeib@gmail.com>

 

**What this PR does / why we need it**:

Not sure how I goofed on this, the sku API pivots on resourceType, not kind. This always was filtering out all resources. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cloud/services/resourceskus/cache.go [hunk 1]
```diff
@@ -99,22 +99,6 @@ func (c *Cache) Get(ctx context.Context, name string, kind Kind) (SKU, error) {
 	return SKU{}, fmt.Errorf("resource sku with name '%s' and category '%s' not found", name, string(kind))
 }
 
-// List returns a list of all known SKUs, useful for further filtering.
-func (c *Cache) List(ctx context.Context, name string, kind Kind) ([]SKU, error) {
```
Comment (alexeldeib) [inline cloud/services/resourceskus/cache.go:103]: unused

**Comment by devigned** [comment]
Just a quick note that the unit tests here were not super helpful to determine the validity of the feature.

**Comment by CecileRobertMichon** [comment]
@devigned good point. Do have ideas around how to make them more helpful or are you saying we should really do #820?

**Comment by devigned** [comment]
One way to make the tests better is record live traffic and then use that in unit tests. I've seen this done multiple times and req / res get stale, test are really brittle, etc... I like the idea of going down that path, but it is a path.

I think the only way we can honestly say a feature works, is to have robust e2e tests that replicate the user experience and are validated through user observable data. Prove it with a test.

**Pull request merged**